presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergFilterPushdown.java

```diff
@@ -0,0 +1,560 @@
+        }
+
+        // Split the filter into 3 groups of conjuncts:
+        //  - range filters that apply to entire columns,
+        //  - range filters that apply to subfields,
+        //  - the rest. Intersect these with possibly pre-existing filters.
+        DomainTranslator.ExtractionResult<Subfield> decomposedFilter = rowExpressionService.getDomainTranslator()
+                .fromPredicate(session, filter, new SubfieldExtractor(functionResolution, rowExpressionService.getExpressionOptimizer(), session).toColumnExtractor());
+        if (currentLayoutHandle.isPresent()) {
+            IcebergTableLayoutHandle currentIcebergLayout = (IcebergTableLayoutHandle) currentLayoutHandle.get();
+            decomposedFilter = intersectExtractionResult(new DomainTranslator.ExtractionResult(currentIcebergLayout.getDomainPredicate(), currentIcebergLayout.getRemainingPredicate()), decomposedFilter);
+        }
+
+        if (decomposedFilter.getTupleDomain().isNone()) {
+            return new ConnectorPushdownFilterResult(EMPTY_TABLE_LAYOUT, FALSE_CONSTANT);
+        }
+
+        RowExpression optimizedRemainingExpression = rowExpressionService.getExpressionOptimizer().optimize(decomposedFilter.getRemainingExpression(), OPTIMIZED, session);
+        if (optimizedRemainingExpression instanceof ConstantExpression) {
+            ConstantExpression constantExpression = (ConstantExpression) optimizedRemainingExpression;
+            if (FALSE_CONSTANT.equals(constantExpression) || constantExpression.getValue() == null) {
+                return new ConnectorPushdownFilterResult(EMPTY_TABLE_LAYOUT, FALSE_CONSTANT);
+            }
+        }
+        Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
+        TupleDomain<ColumnHandle> entireColumnDomain = decomposedFilter.getTupleDomain()
+                .transform(subfield -> isEntireColumn(subfield) ? subfield.getRootName() : null)
+                .transform(columnHandles::get);
+
+        Constraint<ColumnHandle> constraint = new Constraint<>(entireColumnDomain);
+
+        // Extract deterministic conjuncts that apply to partition columns and specify these as Constraint#predicate
+        if (!TRUE_CONSTANT.equals(decomposedFilter.getRemainingExpression())) {
+            LogicalRowExpressions logicalRowExpressions = new LogicalRowExpressions(rowExpressionService.getDeterminismEvaluator(), functionResolution, functionMetadataManager);
+            RowExpression deterministicPredicate = logicalRowExpressions.filterDeterministicConjuncts(decomposedFilter.getRemainingExpression());
+            if (!TRUE_CONSTANT.equals(deterministicPredicate)) {
+                ConstraintEvaluator evaluator = new ConstraintEvaluator(rowExpressionService, session, columnHandles, deterministicPredicate);
+                constraint = new Constraint<>(entireColumnDomain, evaluator::isCandidate);
+            }
+        }
+
+        org.apache.iceberg.Table icebergTable;
```

Comment:
I might have included this change by mistake, it is part of the changes for partitioned tables. icebergTable will be used to get partition columns. 

I will remove this variable from the current commit and it will be added back as part of the partition table changes.